public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -57,7 +57,7 @@ const SubmissionsTable = (): JSX.Element => {
                                 <span className="badge ml-2 float-right"><i className="fa fa-users" title="Group Assignment" /></span> : null}
                         </td>
                         <td><ProgressBar assignmentIndex={assignment.order - 1} courseID={courseID.toString()} submission={submission} type={Progress.OVERVIEW} /></td>
-                        <td>{getFormattedTime(assignment.deadline)}</td>
+                        <td>{assignment.deadline}</td>
                         <td>{deadline.message ? deadline.message : '--'}</td>
```

Comment:
We should move this check and set the `deadline.message` to `--` in `timeFormatter()`. See also check Line 46.

Reply:
That is, it becomes just:
```ts
   <td>{deadline.message}</td>
```

Reply:
I'm not sure I understand the logic on Line 46 well enough to mess with it. Would I need to change the check to `if deadline.message == "--"` ? Or is it ok to always render the rest?